tests/ModelContextProtocol.Tests/Transport/StreamClientTransport.cs

```diff
@@ -11,14 +11,14 @@ internal sealed class StreamClientTransport : TransportBase, IClientTransport
     private readonly JsonSerializerOptions _jsonOptions = McpJsonUtilities.DefaultOptions;
     private Task? _readTask;
     private CancellationTokenSource _shutdownCts = new CancellationTokenSource();
-    private readonly TextReader _stdin;
-    private readonly TextWriter _stdout;
+    private readonly TextReader _stdout;
+    private readonly TextWriter _stdin;
```

Comment:
Why are the names here being swapped?

Reply:
These represent the server's stdin and stdout from the client's perspective. I don't think it's correct to call the server's stdin "stdout" just because we're writing to it. The server's stdin is not any process's stdout under normal circumstances.

Reply:
> These represent the server's stdin and stdout from the client's perspective

Sent us out from the clients perspective? This is the server and the server's state... why would it be looking at its own state from the client's perspective?

Reply:
It's super strange to me to have something named "out" be something you read from and something named "in" be something you write to. If we really want to use that terminology, these should include client in the name to disambiguate.

Reply:
Oh, wait, this is the client. I can read, I swear.

But the point still stands, just inverted. If we want to call the writer stdin and the reader stdout, then we should include in their name server so that it's clear they're referring to the other end of the connection.

Reply:
This is test code, so the names don't matter that much, but the server is the only thing that has a stdin or stdout relevant to the MCP protocol. The server's stdout is not any process's stdin, client or otherwise, so I think calling the TextReader _stdin is just wrong.